### ISSUE TITLE
Sponge not registering Chisel&Bits events.

### ISSUE THREAD

**Issue by TeKGameR950** [SpongePowered/SpongeForge#2167]
**I am currently running**
- SpongeForge version: 1.12.2-2655-7.1.0-BETA-3105
- Forge version: 14.23.3.2655
- Java version: JAVA 8
- Operating System: OMGSERV Hosted srry idk what is it :/

- Plugins/Mods:

**Plugins (12): Minecraft, Minecraft Coder Pack, SpongeAPI, SpongeForge, FlexibleLogin, LuckPerms, Nucleus, PlaceholderAPI, SkinsRestorer, Total Economy, Universe Guard 2, VirtualChest**

**Mods (18): Minecraft, Minecraft Coder Pack, Forge Mod Loader, Minecraft Forge, SpongeAPI, SpongeForge, Chisel, Chisels & Bits, CustomNPCs, Gliby's Voice Chat Mod, HyPlay Mod, Iron Chest, MalisisCore, MalisisDoors, MrCrayfish's Furniture Mod, Obfuscate, Vehicle Mod, WorldEdit**

**Issue Description**

This issue is divided in 2 parts:

1.- If you try to chisel a block inside a protected region (UniverseGuard, FoxGuard, SafeGuard, etc), you will get the drop but the block won't be modified.

2.- If the block inside a protected region IS ALREADY chiseled, you will be able to modify it and you will also get the drop.

Chisel and Bits source: https://github.com/AlgorithmX2/Chisels-and-Bits

SAME ISSUE AS https://github.com/SpongePowered/SpongeForge/issues/765

I created a new issue as someone told me to do that.

Thanks for reading.

**Comment by gabizou** [comment]
Can you try this jar? (I had to zip it so github would allow it). [spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip](https://github.com/SpongePowered/SpongeForge/files/2033026/spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip)
It has some changes to better detect when mods are requesting to modify a block, which Chisel should be able to be blocked correctly from dropping the items. I'll link to the branch when I get a chance. 

Note that this version now requires Forge 2690 at least, but should not affect GP or other protection plugins otherwise.

Adding commit that should fix the issue:
https://github.com/SpongePowered/SpongeForge/commit/315dd62b919b061aeb60bd20c7d187a0b4e29101

**Comment by TeKGameR950** [comment]
I'll try if i find why my server isnt starting with this forge version.

**Comment by gabizou** [comment]
[spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip](https://github.com/SpongePowered/SpongeForge/files/2033186/spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip)
Try this jar now.

**Comment by TeKGameR950** [comment]
Same error after trying again, https://pastebin.com/D91jbdPv

**Comment by gabizou** [comment]
[spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip](https://github.com/SpongePowered/SpongeForge/files/2036195/spongeforge-1.12.2-2690-7.1.0-BETA-0.jar.zip)
Try this one, fixed the crash at startup.

**Comment by TeKGameR950** [comment]
Okay, its better, now the first issue is totally fixed and the second issue is a little bit fixed, what i mean;

Now when you chisel an already chiseled block it doesnt give the microblock but it still modify it, i tried restarting the server/reloading my client to see if it was only modifying the block clientside but no, it really modify it, other players see the modification.

**Comment by gabizou** [comment]
Made a new issue with Chisels and Bits as the block check is passing before sponge can cancel the interaction:
https://github.com/AlgorithmX2/Chisels-and-Bits/issues/390

Otherwise the issue is going to be pending on the mod change.

**Comment by TeKGameR950** [comment]
Already did

He said thats a sponge issue

**Comment by gabizou** [comment]
This is now fixed with the newest versions of Chisels & Bits, latest version can be found here: https://github.com/AlgorithmX2/Chisels-and-Bits/issues/390#issuecomment-394047065
